tests/TestCase/Collection/CollectionTest.php

```diff
@@ -1287,6 +1287,7 @@ public function testCombineWithNonBackedEnum(): void
         ]))->combine('type', 'amount');
     }
 
+    /* Existing behavior is not correct, throws exception in 5.1
```

Comment:
Should this test be executable? It would 'pass' now. I agree that the current output is bizarre. I don't think that having this be a valid scenario is a good one. In the past we've fixed many defects like this. The position in the past has generally been that if your code was relying on defective behavior like this, we're sorry. Keeping this as a 'valid' scenario creates bugs we have to maintain for long periods of time.

Reply:
The 5.1 tests all throw exceptions. I tried to re-create the exception tests with the old behavior and found this was really broken. This was more just calling it out in the review, but the 5.next merge will re-enable the exceptions.

I didn't expect this bug - not sure anyone could be relying on this.